OutOfSchool/OutOfSchool.WebApi.Tests/Services/ChatRoomWorkshopServiceTests.cs

```diff
@@ -448,6 +448,73 @@ public async Task GetUniqueChatRoom_WhenRoomDoesNotExist_ShouldReturnNull()
     }
     #endregion
 
+    #region GetCurrentUserUnreadMessagesCountAsync
+    [Test]
+    public async Task GetCurrentUserUnreadMessagesCount_WhenUserRoleIsParent_ShouldReturnCorrectCount()
+    {
+        // Arrange
+        var parentId = parents[0].Id;
+        var role = Role.Parent;
+        var chatRooms = new List<ChatRoomWorkshopForChatList>
+        {
+            new ChatRoomWorkshopForChatList { NotReadByCurrentUserMessagesCount = 1 },
+            new ChatRoomWorkshopForChatList { NotReadByCurrentUserMessagesCount = 1 },
+        };
+
+        roomWithSpecialModelRepositoryMock
+            .Setup(x => x.GetByParentIdAsync(parentId, It.IsAny<bool>()))
+            .ReturnsAsync(chatRooms);
+
+        // Act
+        var result = await roomService.GetCurrentUserUnreadMessagesCountAsync(parentId, role).ConfigureAwait(false);
+
+        // Assert
+        Assert.IsInstanceOf<int>(result);
```

Comment:
(Excessive code) Please, don't test the compiler. If code changes to other return type you'll see compiler error (where it is valuable) earlier than this unit test will fail.